src/tables/service_periods.py

```diff
@@ -20,3 +23,91 @@ def __init__(self, user=None, passwd=None, hostname="localhost", db_name="apertu
                 ternary SMALLINT NOT NULL CHECK ( (ternary <= 3) AND (ternary >= 1) ),
                 UNIQUE (month, year, ternary)
             );"""])
+
+    
+    def write_table(self, dates):
+        # Dates: list of datetime dates. Only important value in the dates are
+        # month and year.
+        # NOTE: not currently being used but could be useful in the future.
+        data = []
+        for date in dates:
+            data.append([date.month, date.year, self.get_ternary(date.month)])
+        df = pandas.DataFrame(data, columns=self._expected_cols)
+        # service_key is BIGSERIAL so there should never be a conflict.
+        return self._write_table(df)  
+
+
+    def query_month_year(self, month, year):
```

Comment:
I'm still on top of why we have a month column in the service_period table. A service_period spans 4 months (maybe 3 and part of 2 others). We have never gotten confirmation of this information. Why did we choose to have a single month as an attribute for this table?

Reply:
I don't know. We don't have specifications from user so we have to make do with some assumptions. If it turns out that we only use ternary instead of month for the service period, then it's easy to change.